kolibri/core/content/views.py

```diff
@@ -131,14 +137,33 @@ def parse_html(content):
                 )
             },
         )
-        return html5lib.serialize(
+
+        # Currently, html5lib strips the doctype, but it's important for correct rendering, so check the original
+        # content for the doctype and, if found, prepend it to the content serialized by html5lib
+        doctype = None
+        try:
+            encoding = p.documentEncoding
+            if not encoding:
+                encoding = "utf-8"
+            doctype = re.match("(?i)<!DOCTYPE[^<>]*(?:<!ENTITY[^<>]*>[^<>]*)?>", content.decode(encoding))
```

Comment:
curious if you wrote this, or if it came from somewhere? :)

Reply:
Yeah, I googled for some Doctype regexes, and used the one that looked to be the most robust. :)